apis/v1beta1/opentelemetrycollector_types.go

```diff
@@ -94,6 +94,11 @@ type OpenTelemetryCollectorSpec struct {
 	// +required
 	// +kubebuilder:pruning:PreserveUnknownFields
 	Config Config `json:"config"`
+	// ConfigVersions defines the number versions to keep for the collector config. Each config version is stored in a separate ConfigMap.
+	// Defaults to 3. The minimum value is 1.
+	// +optional
+	// +kubebuilder:default:=3
```

Comment:
We can validate the minimum using a validation marker:
```suggestion
	// +kubebuilder:default:=3
	// +kubebuilder:validation:Minimum:=1
```